pkg/master/master.go

```diff
@@ -549,6 +549,46 @@ func (m *Master) InstallSwaggerAPI() {
 	swagger.RegisterSwaggerService(swaggerConfig, m.handlerContainer)
 }
 
+// Picks the best available address for an internal health check
+// TODO: Create utility function, combining this an GetHostIP in kubelet.go?
+func pickAddress(node *api.Node) string {
```

Comment:
A loop with a score function is scary to me. What do you think about building a map and expressing priority based on execution order? Something similar to [GetHostIP()] (https://github.com/GoogleCloudPlatform/kubernetes/blob/4b9a64bcd80cc6ac7ccb3d91606a4bb29f6ab124/pkg/kubelet/kubelet.go#L1716) in the kubelet.